packages/loader/driver-utils/src/mapWithExpiration.ts

```diff
@@ -0,0 +1,115 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+
+/**
+ * An extension of Map that expires (deletes) entries after a period of inactivity.
+ * The policy is based on the last time a key was written to.
+ */
+export class MapWithExpiration<TKey, TValue> extends Map<TKey, TValue> {
```

Comment:
This looks like a very specific implementation for cache expiry for the in-memory cache. Why put this in driver-utils? I would expect this to be part of the in-memory cache. Are you planning on using this elsewhere?
Also, curious why you chose this approach vs implementing it directly in the in-memory cache class.

Reply:
Yeah there are other places that need this kind of thing, e.g. PromiseCache in common-utils.  So I left it as a generic class that could be moved / reused.